src/common/ast/ast.rs

```diff
@@ -0,0 +1,48 @@
+use std::collections::HashMap;
+
+#[derive(Debug, PartialEq)]
+pub struct Model {
+    pub variables: HashMap<Name, DecisionVariable>,
+    pub constraints: Vec<Expression>,
+}
+
+impl Model {
+    // Function to update a DecisionVariable based on its Name
+    pub fn update_domain(&mut self, name: &Name, new_domain: Domain) {
+        if let Some(decision_var) = self.variables.get_mut(name) {
+            decision_var.domain = new_domain;
+        }
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub enum Name {
+    UserName(String),
+    MachineName(i32),
+}
+
+#[derive(Debug, PartialEq)]
+pub struct DecisionVariable {
+    pub domain: Domain,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Domain {
+    BoolDomain,
+    IntDomain(Vec<Range<i32>>),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum Range<A> {
+    Single(A),
+    Bounded(A, A),
```

Comment:
Minor issue, make clear if Bounded(2,5) is {2,3,4,5} or {2,3,4} (I've seen both used).

Reply:
This is a good point, @ozgurakgun what are your thoughts?

Reply:
both sides inclusive was my thinking

Reply:
Should I rename it to `BoundedInclusive` then?

Reply:
There is only one bounded, so I think the name is ok. We want it documented though, definitely.